### PR TITLE
[Integration test for measuring the performance of creating company]LPD-16305 Add an integration test to measure the performance of creat…

### PR DESCRIPTION
**Pull request by slnn**
https://liferay.atlassian.net/browse/LPD-16305

### CODE REVIEW

File: modules/apps/company/company-test/src/testIntegration/java/com/liferay/company/creation/performance/test/CompanyCreationPerformanceTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		new AggregateTestRule(
+			new AssumeTestRule("assume"), new LiferayIntegrationTestRule());
+
+	public static void assume() {
+		Assume.assumeTrue(Validator.isNull(System.getenv("JENKINS_HOME")));
+	}
+
+	@Test
+	public void testSingleThreadAddCompanies() throws Exception {
+		int companiesCount = 10;
```
Comment (dantewang) [inline modules/apps/company/company-test/src/testIntegration/java/com/liferay/company/creation/performance/test/CompanyCreationPerformanceTest.java:50]: @slnn Please make the default count `1`.

File: modules/apps/company/company-test/src/testIntegration/java/com/liferay/company/creation/performance/test/CompanyCreationPerformanceTest.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+		int originalCompaniesCount = _companyLocalService.getCompaniesCount();
+
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				MailMessageListener.class.getName(), LoggerTestUtil.OFF)) {
+
+			long startTime = System.currentTimeMillis();
```
Comment (dantewang) [inline modules/apps/company/company-test/src/testIntegration/java/com/liferay/company/creation/performance/test/CompanyCreationPerformanceTest.java:57]: @slnn Here let's just use `LoggingTimer`.

**Review by dantewang — CHANGES_REQUESTED** [review]

File: modules/apps/company/company-test/src/testIntegration/java/com/liferay/company/creation/performance/test/CompanyCreationPerformanceTest.java (hunk 2 above)
Comment (slnn) [reply modules/apps/company/company-test/src/testIntegration/java/com/liferay/company/creation/performance/test/CompanyCreationPerformanceTest.java:51]: Done

File: modules/apps/company/company-test/src/testIntegration/java/com/liferay/company/creation/performance/test/CompanyCreationPerformanceTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+	public static void assume() {
+		Assume.assumeTrue(Validator.isNull(System.getenv("JENKINS_HOME")));
+	}
+
+	@Test
+	public void testSingleThreadAddCompanies() throws Exception {
+		int originalCompaniesCount = _companyLocalService.getCompaniesCount();
+
+		try (LoggingTimer loggingTimer = new LoggingTimer()) {
```
Comment (dantewang) [inline modules/apps/company/company-test/src/testIntegration/java/com/liferay/company/creation/performance/test/CompanyCreationPerformanceTest.java:50]: @slnn With LoggingTimer, you don't need to manually record the time. You remove the time related logic, and move the asserting out of the `try() {}` block.

Comment (slnn) [reply modules/apps/company/company-test/src/testIntegration/java/com/liferay/company/creation/performance/test/CompanyCreationPerformanceTest.java:50]: Oh, then we can use 20794 ms in the output as the result, right?

` 2024-02-01 08:35:59.873 INFO  [com.liferay.company.test-executor-thread][LoggingTimer:35] Completed com.liferay.company.creation.performance.test.CompanyCreationPerformanceTest#testSingleThreadAddCompanies in 20794 ms`

Comment (slnn) [reply modules/apps/company/company-test/src/testIntegration/java/com/liferay/company/creation/performance/test/CompanyCreationPerformanceTest.java:49]: Done

**Review by dantewang — APPROVED** [review]

**Comment by dantewang** [comment]
ci:test:sf

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
See https://github.com/liferay-core-infra/liferay-portal/pull/6651

**Pull request closed without merging**